torch/functional.py

```diff
@@ -934,6 +935,65 @@ def btrisolve(b, LU_data, LU_pivots, out=None):
                   "removed in the next release. Please use torch.lu_solve instead.", stacklevel=2)
     return torch.lu_solve(b, LU_data=LU_data, LU_pivots=LU_pivots, out=out)
 
+def nonzero(input, **kwargs):
+    r"""
+    nonzero(input, *, out=None, as_tuple=False) -> LongTensor or tuple of LongTensors
```

Comment:
What do you think about doing all the parsing at the python arg parser level?  That way we are guaranteed to treat everything consistently.

Reply:
Right now there is no support there afaik for returning different result types depending on the value of a kwarg, so it would have to be significantly refactored.

Reply:
can't you just handwrite the parsing code?  At the end of the day, the parsing related code just returns a PyObject, so this should be fine, no?